src/libsyntax/tokenstream.rs

```diff
@@ -820,98 +929,108 @@ impl Index<usize> for TokenStream {
     type Output = TokenTree;
 
     fn index(&self, index: usize) -> &TokenTree {
```

Comment:
If `.iter().nth(i)` works and indexing is undesirable by default, I'm not sure why we need/want it.
Not to mention the need for a `&Self::Target` which would limit our ability to create some of the values on-demand (if the internal representation doesn't match the one in the stable API).